tests/Properties.hs

```diff
@@ -76,20 +77,15 @@ prop_unsafeIndexBB xs =
 prop_bijectionBB  (Char8 c) = (P.w2c . P.c2w) c == id c
 prop_bijectionBB'        w  = (P.c2w . P.w2c) w == id w
 
-prop_head2BB xs    = (not (null xs)) ==> head xs   == (P.unsafeHead . P.pack) xs
-
-prop_tail1BB xs    = (not (null xs)) ==> tail xs    == (P.unpack . P.unsafeTail. P.pack) xs
-
-prop_last1BB xs    = (not (null xs)) ==> last xs    == (P.unsafeLast . P.pack) xs
-
-prop_init1BB xs     =
-    (not (null xs)) ==>
-    init xs    == (P.unpack . P.unsafeInit . P.pack) xs
+prop_unsafeHead xs = not (P.null xs) ==> P.head xs === P.unsafeHead xs
+prop_unsafeTail xs = not (P.null xs) ==> P.tail xs === P.unsafeTail xs
+prop_unsafeLast xs = not (P.null xs) ==> P.last xs === P.unsafeLast xs
+prop_unsafeInit xs = not (P.null xs) ==> P.init xs === P.unsafeInit xs
```

Comment:
My first suspicion was that these tests would be pretty useless, because I thought that `head` is implemented in terms of `unsafeHead`. But this is not the case (yet).